src/api/java/CMakeLists.txt

```diff
@@ -2,28 +2,93 @@
 ## directory for licensing information.
 ##
 
-# create a directory for the cvc package
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/cvc")
+# create directories
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc5/")
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/jni")
 
-# Generate cvc/Kind.java
+# Generate cvc5/Kind.java
 configure_file(genkinds.py.in genkinds.py)
 add_custom_target(
   gen-java-kinds
   ALL
   COMMAND
-    "${PYTHON_EXECUTABLE}"
-    "${CMAKE_CURRENT_BINARY_DIR}/genkinds.py"
-    --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cpp/cvc5_kind.h"
-    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/cvc/Kind"
+  "${PYTHON_EXECUTABLE}"
+  "${CMAKE_CURRENT_BINARY_DIR}/genkinds.py"
+  --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cpp/cvc5_kind.h"
+  --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc5/Kind"
   DEPENDS
-    "${CMAKE_CURRENT_BINARY_DIR}/genkinds.py"
+  "${CMAKE_CURRENT_BINARY_DIR}/genkinds.py"
   COMMENT
-    "Generate Kind.java"
+  "Generate Kind.java"
 )
+
+# find java
+find_package(Java COMPONENTS Development REQUIRED)
+include(UseJava)
+
+# specify java source files
+set(java_files
+  ${CMAKE_CURRENT_LIST_DIR}/cvc5/CVC5ApiException.java
+  ${CMAKE_CURRENT_LIST_DIR}/cvc5/CVC5ApiRecoverableException.java
+  ${CMAKE_CURRENT_LIST_DIR}/cvc5/IPointer.java
+  ${CMAKE_CURRENT_LIST_DIR}/cvc5/Solver.java
+  ${CMAKE_CURRENT_LIST_DIR}/cvc5/Utils.java
+  ${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc5/Kind.java
+)
+
+# generate jni headers
+add_custom_target(
+  generate-jni-headers
+  ALL
+  COMMAND
+  # generate jni header files
+  ${Java_JAVAC_EXECUTABLE} -h jni ${java_files} -d jni
+  COMMENT "Generate jni headers"
+  VERBATIM
+)
+
+add_dependencies(generate-jni-headers gen-java-kinds)
+
+# find jni package
+find_package(JNI REQUIRED)
+message(STATUS "JAVA_AWT_LIBRARY     : ${JAVA_AWT_LIBRARY}")
+message(STATUS "JNI_INCLUDE_DIRS     : ${JNI_INCLUDE_DIRS}")
+message(STATUS "JNI_LIBRARIES        : ${JNI_LIBRARIES} ")
+message(STATUS "JNI_FOUND            : ${JNI_FOUND}")
+message(STATUS "JAVA_AWT_LIBRARY     : ${JAVA_AWT_LIBRARY}")
+message(STATUS "JAVA_JVM_LIBRARY     : ${JAVA_JVM_LIBRARY}")
+message(STATUS "JAVA_INCLUDE_PATH    : ${JAVA_INCLUDE_PATH}")
+message(STATUS "JAVA_INCLUDE_PATH2   : ${JAVA_INCLUDE_PATH2}")
+message(STATUS "JAVA_AWT_INCLUDE_PATH: ${JAVA_AWT_INCLUDE_PATH}")
+
+add_library(cvc5jni SHARED jni/cvc5_Solver.cpp)
+add_dependencies(cvc5jni generate-jni-headers)
+
+target_include_directories(cvc5jni PUBLIC ${JNI_INCLUDE_DIRS})
+target_include_directories(cvc5jni PUBLIC ${PROJECT_SOURCE_DIR}/src)
+target_include_directories(cvc5jni PUBLIC ${CMAKE_BINARY_DIR}/src/)
+target_include_directories(cvc5jni PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/jni)
+target_link_libraries(cvc5jni ${JNI_LIBRARIES})
+target_link_libraries(cvc5jni cvc4)
+
+set(CVC5_JAR "cvc5-${CVC4_MAJOR}.${CVC4_MINOR}.${CVC4_RELEASE}.jar" CACHE STRING "cvc5 jar" FORCE)
```

Comment:
This variable seems to be overwritten in line 93.

Reply:
Fixed by introducing a new new variable `CVC5_JAR_PATH`.